### PR TITLE
Preserving the association conditions when using ExistsIn

### LINKED ISSUE #7968: ExistsIn should apply association conditions
I'm using an `ExistsIn` build rule with one of my association which has conditions. Since I'm explicitly checking for the existence in an association - not in a table - I would expect `ExistsIn` to use the association conditions to check for the existence.

My suggestion would be respecting the association conditions, so data integrity will be enforced. If there are conditions on an association which shouldn't be checked (I can't think of a reason for that, to be honest), you could still pass an instance of a table instead of the association name.

What are the opinions on this matter? I would be happy to implement this if you agree on this.

### PR DESCRIPTION
**Pull request by lorenzo**
Closes #7968

### CODE REVIEW

**Comment by GuidoHendriks** [comment]
Awesome! 👌🏻

File: src/ORM/Association.php [hunk 1]
```diff
@@ -679,6 +679,25 @@ public function find($type = null, array $options = [])
     }
 
     /**
+     * Proxies the operation to the target table's exists method after
+     * appending the default conditions for thisassociation
```
Comment (markstory) [inline src/ORM/Association.php]: Need an extra space in `thisassociation`.

File: src/ORM/Rule/ExistsIn.php [hunk 2]
```diff
@@ -77,22 +77,20 @@ public function __invoke(EntityInterface $entity, array $options)
...
-        if ($target instanceof Association) {
-            $bindingKey = (array)$target->bindingKey();
-            $target = $target->target();
-        } else {
-            $bindingKey = (array)$target->primaryKey();
-        }
-
-        if (!empty($options['_sourceTable']) && $target === $options['_sourceTable']) {
-            return true;
-        }
```
Comment (markstory) [inline src/ORM/Rule/ExistsIn.php]: Nice :+1:

**Comment by lorenzo** [comment]
@markstory Fixed typo

File: tests/TestCase/ORM/RulesCheckerIntegrationTest.php [hunk 3]
```diff
@@ -860,4 +860,28 @@ public function testAvoidExistsInOnAutomaticSaving()
...
+    public function testExistsInDomainRuleWithAssociationConditions()
+    {
+        $entity = new Entity([
+            'title' => 'An Article',
+            'author_id' => 1
+        ]);
+
+        $table = TableRegistry::get('Articles');
+        $table->belongsTo('Authors', [
+            'conditions' => ['Authors.name !=' => 'mariano']
```
Comment (jhli) [inline tests/TestCase/ORM/RulesCheckerIntegrationTest.php]: A test that fails, too?

Comment (jhli) [inline tests/TestCase/ORM/RulesCheckerIntegrationTest.php]: ^ ignore

**Pull request merged**